receiver/azureblobreceiver/config.go

```diff
@@ -3,10 +3,44 @@
 
 package azureblobreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureblobreceiver"
 
+import (
+	"errors"
+	"fmt"
+
+	"go.uber.org/multierr"
+)
+
+const (
+	azureCloud           = "AzureCloud"
+	azureGovernmentCloud = "AzureUSGovernment"
+)
+
+var (
+	// Predefined error responses for configuration validation failures
+	errMissingTenantID          = errors.New(`"TenantID" is not specified in config`)
+	errMissingClientID          = errors.New(`"ClientID" is not specified in config`)
+	errMissingClientSecret      = errors.New(`"ClientSecret" is not specified in config`)
+	errMissingStorageAccountURL = errors.New(`"StorageAccountURL" is not specified in config`)
+	errMissingConnectionString  = errors.New(`"ConnectionString" is not specified in config`)
+	errInvalidCloud             = errors.New(`"Cloud" is invalid`)
+)
+
 type Config struct {
+	// Type of authentication to use
+	Authentication string `mapstructure:"auth"`
 	// Azure Blob Storage connection key,
 	// which can be found in the Azure Blob Storage resource on the Azure Portal. (no default)
 	ConnectionString string `mapstructure:"connection_string"`
+	// Storage Account URL, used with Service Principal authentication
+	StorageAccountURL string `mapstructure:"storage_account_url"`
+	// Tenant ID, used with Service Principal authentication
+	TenantID string `mapstructure:"tenant_id"`
+	// Client ID, used with Service Principal authentication
+	ClientID string `mapstructure:"client_id"`
+	// Client secret, used with Service Principal authentication
+	ClientSecret string `mapstructure:"client_secret"`
```

Comment:
Sensitive configuration fields should use `configopaque.String`
```suggestion
	ClientSecret configopaque.String `mapstructure:"client_secret"`
```